From bc8799d7d7f240887086a16e8a400e049ea64df0 Mon Sep 17 00:00:00 2001
From: "Jasper St. Pierre" <jstpierre@mecheye.net>
Date: Mon, 5 May 2014 19:06:47 -0400
Subject: [PATCH] constraints: Complete fix for size hints constrainment

---
 src/core/constraints.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/core/constraints.c b/src/core/constraints.c
index df9c219..bfc7090 100644
--- a/src/core/constraints.c
+++ b/src/core/constraints.c
@@ -950,6 +950,14 @@ constrain_size_increments (MetaWindow         *window,
   if (new_height < window->size_hints.min_height)
     new_height += ((window->size_hints.min_height - new_height)/hi + 1)*hi;
 
+  {
+    client_rect.width = new_width;
+    client_rect.height = new_height;
+    meta_window_client_rect_to_frame_rect (window, &client_rect, &client_rect);
+    new_width = client_rect.width;
+    new_height = client_rect.height;
+  }
+
   /* Figure out what original rect to pass to meta_rectangle_resize_with_gravity
    * See bug 448183
    */
-- 
1.9.0

