diff -PruN metacity-2.12.1.old/src/display.c metacity-2.12.1/src/display.c
--- metacity-2.12.1.old/src/display.c	2005-10-03 19:57:32.000000000 +0200
+++ metacity-2.12.1/src/display.c	2005-10-09 16:12:46.000000000 +0200
@@ -1641,7 +1641,8 @@
                    * in click to focus mode? The debate rages.
                    * Feel free to change TRUE to FALSE or vice versa
                    */
-                  if (TRUE /* meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK */) 
+                  /* if (TRUE /* meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK )*/ 
+		  if(meta_prefs_get_raise_on_click ())
                     meta_window_raise (window);
                   else
                     meta_topic (META_DEBUG_FOCUS,
diff -PruN metacity-2.12.1.old/src/metacity.schemas.in metacity-2.12.1/src/metacity.schemas.in
--- metacity-2.12.1.old/src/metacity.schemas.in	2005-09-29 17:53:52.000000000 +0200
+++ metacity-2.12.1/src/metacity.schemas.in	2005-10-09 16:12:46.000000000 +0200
@@ -95,6 +95,20 @@
     </schema>
 
     <schema>
+      <key>/schemas/apps/metacity/general/raise_on_click</key>
+      <applyto>/apps/metacity/general/raise_on_click</applyto>
+      <owner>metacity</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+         <short>Raise current window if click event occurs</short>
+         <long>
+          If true the focused window will be automatically raised.
+         </long>
+      </locale>
+    </schema>
+
+    <schema>
       <key>/schemas/apps/metacity/general/auto_raise_delay</key>
       <applyto>/apps/metacity/general/auto_raise_delay</applyto>
       <owner>metacity</owner>
diff -PruN metacity-2.12.1.old/src/prefs.c metacity-2.12.1/src/prefs.c
--- metacity-2.12.1.old/src/prefs.c	2005-09-29 17:53:52.000000000 +0200
+++ metacity-2.12.1/src/prefs.c	2005-10-09 16:14:26.000000000 +0200
@@ -53,6 +53,7 @@
 #define KEY_BUTTON_LAYOUT "/apps/metacity/general/button_layout"
 #define KEY_REDUCED_RESOURCES "/apps/metacity/general/reduced_resources"
 #define KEY_GNOME_ACCESSIBILITY "/desktop/gnome/interface/accessibility"
+#define KEY_RAISE_ON_CLICK "/apps/metacity/general/raise_on_click"
 
 #define KEY_COMMAND_PREFIX "/apps/metacity/keybinding_commands/command_"
 
@@ -92,6 +93,7 @@
 static gboolean bell_is_audible = TRUE;
 static gboolean reduced_resources = FALSE;
 static gboolean gnome_accessibility = FALSE;
+static gboolean raise_on_click = FALSE;
 static char *cursor_theme = NULL;
 static int   cursor_size = 24;
 
@@ -132,6 +134,7 @@
 static gboolean update_action_double_click_titlebar (const char *value);
 static gboolean update_auto_raise          (gboolean   value);
 static gboolean update_auto_raise_delay    (int        value);
+static gboolean update_raise_on_click     (gboolean    value);
 static gboolean update_button_layout      (const char *value);
 static gboolean update_window_binding     (const char *name,
                                            const char *value);
@@ -307,7 +310,7 @@
   char *str_val;
   int int_val;
   gboolean bool_val, bool_val_2;
-  
+   
   if (default_client != NULL)
     return;
   
@@ -363,7 +366,11 @@
   cleanup_error (&err);
   update_auto_raise_delay (int_val);
   
-
+  bool_val = gconf_client_get_bool (default_client, KEY_RAISE_ON_CLICK,
+                 &err);
+  cleanup_error (&err);
+  update_raise_on_click (bool_val);                 
+  
   str_val = gconf_client_get_string (default_client, KEY_THEME,
                                      &err);
   cleanup_error (&err);
@@ -732,6 +739,22 @@
         queue_changed (META_PREF_AUTO_RAISE_DELAY);
     
     }
+  else if (strcmp (key, KEY_RAISE_ON_CLICK) == 0)
+    {
+      gboolean b;
+
+      if (value && value->type != GCONF_VALUE_BOOL)
+        {
+          meta_warning (_("GConf key \"%s\" is set to an invalid type\n"),
+                        KEY_RAISE_ON_CLICK);
+          goto out;
+        }
+
+      b = value ? gconf_value_get_bool (value) : raise_on_click;
+
+      if (update_raise_on_click (b))
+        queue_changed (META_PREF_RAISE_ON_CLICK);
+    }    
   else if (str_has_prefix (key, KEY_COMMAND_PREFIX))
     {
       const char *str;
@@ -1443,6 +1466,16 @@
   return old != auto_raise;
 }
 
+static gboolean
+update_raise_on_click (gboolean value)
+{
+  gboolean old = raise_on_click;
+
+  raise_on_click = value;
+
+  return old != raise_on_click;
+}
+
 #define MAX_REASONABLE_AUTO_RAISE_DELAY 10000
   
 static gboolean
@@ -1523,6 +1556,9 @@
     case META_PREF_AUTO_RAISE:
       return "AUTO_RAISE";
       
+    case META_PREF_RAISE_ON_CLICK:
+      return "RAISE_ON_CLICK";
+      
     case META_PREF_AUTO_RAISE_DELAY:
       return "AUTO_RAISE_DELAY";
 
@@ -2259,6 +2295,12 @@
   return auto_raise;
 }
 
+gboolean
+meta_prefs_get_raise_on_click ()
+{
+	return raise_on_click;
+}
+
 int
 meta_prefs_get_auto_raise_delay (void)
 {
diff -PruN metacity-2.12.1.old/src/prefs.h metacity-2.12.1/src/prefs.h
--- metacity-2.12.1.old/src/prefs.h	2005-09-29 17:53:52.000000000 +0200
+++ metacity-2.12.1/src/prefs.h	2005-10-09 16:15:29.000000000 +0200
@@ -50,7 +50,8 @@
   META_PREF_REDUCED_RESOURCES,
   META_PREF_GNOME_ACCESSIBILITY,
   META_PREF_CURSOR_THEME,
-  META_PREF_CURSOR_SIZE
+  META_PREF_CURSOR_SIZE,
+  META_PREF_RAISE_ON_CLICK
 } MetaPreference;
 
 typedef void (* MetaPrefsChangedFunc) (MetaPreference pref,
@@ -76,6 +77,7 @@
 int                         meta_prefs_get_auto_raise_delay   (void);
 gboolean                    meta_prefs_get_reduced_resources  (void);
 gboolean                    meta_prefs_get_gnome_accessibility (void);
+gboolean                    meta_prefs_get_raise_on_click     (void);
 
 const char*                 meta_prefs_get_command            (int i);
 
