--- metacity-2.8.1/configure.in.orig	2004-05-05 21:26:39.000000000 +0200
+++ metacity-2.8.1/configure.in	2004-05-05 21:27:14.000000000 +0200
@@ -242,9 +242,9 @@
         *)
             # Check for XFree
             use_xfree_xinerama=yes
-            AC_CHECK_LIB(Xinerama, XineramaQueryExtension,
+	    AC_CHECK_LIB(Xinerama_pic, XineramaQueryExtension,
                 [AC_CHECK_HEADER(X11/extensions/Xinerama.h,
-                                 X_EXTRA_LIBS="-lXinerama $X_EXTRA_LIBS"   
+				 X_EXTRA_LIBS="-Wl,-lXinerama_pic $X_EXTRA_LIBS"
                                  if test -z "`echo $ALL_X_LIBS | grep "\-lXext" 2> /dev/null`"; then
                                      X_EXTRA_LIBS="-lXext $X_EXTRA_LIBS"
                                  fi
--- metacity-2.8.1/configure.orig	2004-05-05 21:28:22.000000000 +0200
+++ metacity-2.8.1/configure	2004-05-05 21:32:06.000000000 +0200
@@ -26160,13 +26160,13 @@
         *)
             # Check for XFree
             use_xfree_xinerama=yes
-            echo "$as_me:$LINENO: checking for XineramaQueryExtension in -lXinerama" >&5
-echo $ECHO_N "checking for XineramaQueryExtension in -lXinerama... $ECHO_C" >&6
-if test "${ac_cv_lib_Xinerama_XineramaQueryExtension+set}" = set; then
+	    echo "$as_me:$LINENO: checking for XineramaQueryExtension in -lXinerama_pic" >&5
+echo $ECHO_N "checking for XineramaQueryExtension in -lXinerama_pic... $ECHO_C" >&6
+if test "${ac_cv_lib_Xinerama_pic_XineramaQueryExtension+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lXinerama -lXext $ALL_X_LIBS $LIBS"
+LIBS="-lXinerama_pic -lXext $ALL_X_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -26211,20 +26211,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_Xinerama_XineramaQueryExtension=yes
+  ac_cv_lib_Xinerama_pic_XineramaQueryExtension=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_Xinerama_XineramaQueryExtension=no
+ac_cv_lib_Xinerama_pic_XineramaQueryExtension=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_Xinerama_XineramaQueryExtension" >&5
-echo "${ECHO_T}$ac_cv_lib_Xinerama_XineramaQueryExtension" >&6
-if test $ac_cv_lib_Xinerama_XineramaQueryExtension = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_Xinerama_pic_XineramaQueryExtension" >&5
+echo "${ECHO_T}$ac_cv_lib_Xinerama_pic_XineramaQueryExtension" >&6
+if test $ac_cv_lib_Xinerama_pic_XineramaQueryExtension = yes; then
   echo "$as_me:$LINENO: checking for X11/extensions/Xinerama.h" >&5
 echo $ECHO_N "checking for X11/extensions/Xinerama.h... $ECHO_C" >&6
 if test "${ac_cv_header_X11_extensions_Xinerama_h+set}" = set; then
@@ -26274,7 +26274,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_header_X11_extensions_Xinerama_h" >&5
 echo "${ECHO_T}$ac_cv_header_X11_extensions_Xinerama_h" >&6
 if test $ac_cv_header_X11_extensions_Xinerama_h = yes; then
-  X_EXTRA_LIBS="-lXinerama $X_EXTRA_LIBS"
+  X_EXTRA_LIBS="-Wl,-lXinerama_pic $X_EXTRA_LIBS"
                                  if test -z "`echo $ALL_X_LIBS | grep "\-lXext" 2> /dev/null`"; then
                                      X_EXTRA_LIBS="-lXext $X_EXTRA_LIBS"
                                  fi
