Index: src/display.c
===================================================================
RCS file: /cvs/gnome/metacity/src/display.c,v
retrieving revision 1.211
diff -p -u -r1.211 display.c
--- metacity-2.6.3/src/display.c	26 Dec 2003 02:42:38 -0000	1.211
+++ metacity-2.6.3/src/display.c	5 Jan 2004 04:15:39 -0000
@@ -1371,7 +1371,7 @@ event_callback (XEvent   *event,
 
                   /* Raise on clicking the client area always or only
                    * in click to focus mode? The debate rages.
-                   * Feel free to change TRUE to FALSE
+                   * Feel free to change TRUE to FALSE or vice versa
                    */
                   if (TRUE /* meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK */) 
                     {
@@ -3303,6 +3303,8 @@ meta_display_grab_focus_window_button (M
   /* Grab button 1 for activating unfocused windows */
   meta_verbose ("Grabbing unfocused window buttons for %s\n", window->desc);
 
+#if 0
+  /* FIXME:115072 */
   /* Don't grab at all unless in click to focus mode. In click to
    * focus, we may sometimes be clever about intercepting and eating
    * the focus click. But in mouse focus, we never do that since the
@@ -3314,6 +3316,7 @@ meta_display_grab_focus_window_button (M
       meta_verbose (" (well, not grabbing since not in click to focus mode)\n");
       return;
     }
+#endif
   
   if (window->have_focus_click_grab)
     {
Index: src/window.c
===================================================================
RCS file: /cvs/gnome/metacity/src/window.c,v
retrieving revision 1.319
diff -p -u -r1.319 window.c
--- metacity-2.6.3/src/window.c	21 Dec 2003 06:31:40 -0000	1.319
+++ metacity-2.6.3/src/window.c	5 Jan 2004 04:16:00 -0000
@@ -4118,7 +4118,8 @@ meta_window_notify_focus (MetaWindow *wi
            * with some things you might do inside the focused window, by
            * causing the client to get funky enter/leave events.
            */
-          meta_display_ungrab_focus_window_button (window->display, window);
+          if (meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK)
+            meta_display_ungrab_focus_window_button (window->display, window);
         }
     }
   else if (event->type == FocusOut ||
@@ -4157,8 +4158,9 @@ meta_window_notify_focus (MetaWindow *wi
           meta_window_update_layer (window);
 
           /* Re-grab for click to focus, if necessary */
-          meta_display_grab_focus_window_button (window->display, window);
-        }
+          if (meta_prefs_get_focus_mode () == META_FOCUS_MODE_CLICK)
+            meta_display_grab_focus_window_button (window->display, window);
+       }
     }
 
   /* Now set _NET_ACTIVE_WINDOW hint */
