Index: mutter-2.31.5/configure.in
===================================================================
--- mutter-2.31.5.orig/configure.in	2010-07-12 12:40:34.000000000 -0300
+++ mutter-2.31.5/configure.in	2010-07-13 11:46:01.276640551 -0300
@@ -108,11 +108,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
