Index: mutter-3.2.1/configure.in
===================================================================
--- mutter-3.2.1.orig/configure.in	2011-10-18 05:17:13.000000000 +0200
+++ mutter-3.2.1/configure.in	2011-11-19 01:00:36.462920873 +0100
@@ -477,11 +477,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
