Index: mutter-3.16.3/configure.ac
===================================================================
--- mutter-3.16.3.orig/configure.ac	2015-07-02 18:53:12.248070631 +0200
+++ mutter-3.16.3/configure.ac	2015-07-02 18:53:12.248070631 +0200
@@ -379,11 +379,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
