Index: mutter-3.21.92/configure.ac
===================================================================
--- mutter-3.21.92.orig/configure.ac	2016-09-13 17:13:17.951200746 +0200
+++ mutter-3.21.92/configure.ac	2016-09-13 17:13:17.947200734 +0200
@@ -425,11 +425,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
