Index: mutter-3.20.1/configure.ac
===================================================================
--- mutter-3.20.1.orig/configure.ac	2016-04-16 23:58:51.024922925 +0200
+++ mutter-3.20.1/configure.ac	2016-04-16 23:58:51.020922863 +0200
@@ -376,11 +376,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
