Index: mutter-3.18.1/configure.ac
===================================================================
--- mutter-3.18.1.orig/configure.ac	2015-10-16 22:32:21.515230193 +0200
+++ mutter-3.18.1/configure.ac	2015-10-16 22:32:21.511229888 +0200
@@ -371,11 +371,6 @@
     esac
 
     case " $CFLAGS " in
-    *[\ \	]-Wcast-align[\ \	]*) ;;
-    *) CFLAGS="$CFLAGS -Wcast-align" ;;
-    esac
-
-    case " $CFLAGS " in
     *[\ \	]-Wsign-compare[\ \	]*) ;;
     *) CFLAGS="$CFLAGS -Wsign-compare" ;;
     esac
